codegen/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/samples/KDocSamplesGenerator.kt

```diff
@@ -0,0 +1,143 @@
+ * and pre-processes the documentation to insert references to the generated sample identifiers for operations
+ * that have [examples](https://smithy.io/2.0/spec/documentation-traits.html#smithy-api-examples-trait)
+ */
+class KDocSamplesGenerator : KotlinIntegration {
+    override fun enabledForService(model: Model, settings: KotlinSettings): Boolean {
+        val topDownIndex = TopDownIndex.of(model)
+        val operations = topDownIndex.getContainedOperations(settings.service)
+        return operations.any { it.hasTrait<ExamplesTrait>() }
+    }
+
+    /**
+     * This should run _after_ [software.amazon.smithy.kotlin.codegen.lang.DocumentationPreprocessor]
+     */
+    override fun preprocessModel(model: Model, settings: KotlinSettings): Model {
+        val transformer = ModelTransformer.create()
+        return transformer.mapTraits(model) { shape, trait ->
+            when {
+                shape is OperationShape && shape.hasTrait<ExamplesTrait>() && trait is DocumentationTrait -> {
+                    val examplesTrait = shape.expectTrait<ExamplesTrait>()
+                    val filtered = examplesTrait.examples.filterNot { it.error.isPresent }
+                    val kdocSampleIdentifiers = filtered.indices.joinToString(separator = "\n") { idx ->
+                        val identifier = sampleIdentifier(settings, shape, idx)
+                        "@sample $identifier"
+                    }
+
+                    val updatedDocs = trait.value + "\n\n" + kdocSampleIdentifiers
+                    DocumentationTrait(updatedDocs, trait.sourceLocation)
+                }
+                else -> trait
+            }
+        }
+    }
+
+    private fun sampleIdentifier(settings: KotlinSettings, op: OperationShape, index: Int): String =
+        listOf(
+            samplePackage(settings),
+            sampleClassName(op),
+            sampleFunctionName(index),
+        ).joinToString(separator = ".")
+
+    private fun sampleFunctionName(index: Int): String = "sample${index + 1}"
+    private fun sampleClassName(op: OperationShape): String = op.id.name
+
+    private fun samplePackage(settings: KotlinSettings): String = settings.pkg.subpackage("samples")
+    override fun writeAdditionalFiles(ctx: CodegenContext, delegator: KotlinDelegator) {
+        val topDownIndex = TopDownIndex.of(ctx.model)
+        val operations = topDownIndex.getContainedOperations(ctx.settings.service)
+        operations.filter { it.hasTrait<ExamplesTrait>() }
+            .forEach { op ->
+                val examples = op.expectTrait<ExamplesTrait>()
+
+                val writer = KotlinWriter(samplePackage(ctx.settings))
+                writer.withBlock("class #L {", "}", sampleClassName(op)) {
+                    examples.examples
+                        // unclear what benefit error examples provide, omit for now
+                        .filterNot { it.error.isPresent }
```

Comment:
**Style**: Fluent call chains should have every `.` invocation on a subsequent line, including the first one:

```kotlin
examples
    .examples
    .filterNot { is.error.isPresent }
    …
```

Same comment below about `example.documentation`.